kratos/mpi/includes/mpi_data_communicator.h

```diff
@@ -428,8 +428,8 @@ class MPIDataCommunicator: public DataCommunicator
         MPI_Op Operation) const;
 
     template<class TDataType> void SendRecvDetail(
-        const TDataType& rSendMessage, const int SendDestination,
-        TDataType& rRecvMessage, const int RecvSource) const;
+        const TDataType& rSendMessage, const int SendDestination, const int SendTag,
```

Comment:
just wondering:
would it make sense to add defaults to the tags as:

     void SendRecv(
         const std::string& rSendValues, const int SendDestination,
         const std::string& rSendValues, const int SendDestination, const int SendTag = SendDestination,
         std::string& rRecvValues, const int RecvSource) const override;
         std::string& rRecvValues, const int RecvSource, const int RecvTag = RecvSource) const override;

Reply:
I do not know what would be a good default value for the tags. The only relevant use of them I have found so far is the communicator, where we use the color_id as a tag to differentiate between successive color stages, but I do not have this information here.

Using the sender_id as a tag would work except if there are multiple send-recv among the same pair of nodes in a given algorithm, which I guess should be rare enough?

Reply:
thinking about, it may well be pretty common. just sync you need to send the values of PRESSURE and DISPLACEMENT_X for any reason. you would need to package it differently (most probably) because one is a variable and the other one a variable component

Reply:
i mean, this is not a problem with blocking calls, but it may well be problematic with non blocking versions. I don't have a clue on how to tag this univocally.

of course what i am saying is actually telling that my comment is not relevant :-D, nevertheless i don't know how we will have to use your version in the practice :-(

Reply:
Yes, I'm afraid that in non-trivial cases only the caller knows what is a good value for this, so we need to leave it as an argument... Maybe once we have practical examples we can figure out a smarter way to do it.